File: backend/src/main/kotlin/petanitambak/petanitambakbackend/model/Activity.kt
package petanitambak.petanitambakbackend.model

import jakarta.persistence.*

@Entity
data class Activity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Int? = null,

    var name: String? = null,

    val details: String? = null,

    @ManyToOne(optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    var category: Category? = null

)

File: frontend/modules/monitoring/lib/domain/entities/activity.dart
import 'package:equatable/equatable.dart';

class Activity extends Equatable {
  String name;
  String details;

  Activity({
    required this.name,
    required this.details,
  });

  @override
  List<Object?> get props => [
        name,
        details,
      ];
}
